### PR TITLE
Refactor to implement classes

### PR DESCRIPTION
**Pull request by nachosala89**
- Refactored objects and arrays to be implemented as classes.
- Refactored functions to be implemented as class methods.

### CODE REVIEW

File: js/dom.js [hunk 1]
```diff
@@ -1,57 +1,82 @@
...
 const authorIn = document.querySelector('#author');
 const button = document.querySelector('#add-btn');
 
-button.addEventListener('click', () => { addBook(books, titleIn.value, authorIn.value); });
+button.addEventListener('click', () => { createArticle(books, titleIn.value, authorIn.value); });
 
 if (localStorage.getItem('books') !== null) {
   window.addEventListener('load', () => {
-    books = JSON.parse(localStorage.getItem('books'));
+    const storedList = JSON.parse(localStorage.getItem('books'));
```
Comment (Davidosky007) [inline js/dom.js:79]: * I get this error when i add a new book kindly 🙏 fix it 
image

File: js/dom.js [hunk 2]
```diff
@@ -1,57 +1,82 @@
...
 
-function displayBooksList(books) {
-  for (let i = 0; i < books.length; i += 1) {
-    createArticle(books, i);
+function createArticle(books, title, author) {
+  if (title !== '' && author !== '') {
+    const book = new Book(books.length, title, author);
+    books.addBook(book);
+    displayArticle(book);
+    localStorage.setItem('books', JSON.stringify(book.list));
```
Comment (Davidosky007) [inline js/dom.js]: * I see undefined in the local storage when i add books kindly look into this 🙏 
image

**Review by Davidosky007 — CHANGES_REQUESTED** [review]
## Hi @Fanger53 @nachosala89 👋, 

Good job so far! 💪 
* Nice looking readme file ✔️ 
* Descriptive commit messages ✔️ 
* No linter errors ✔️ 

There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
* Kindly address the comments below 🙏 
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- favour the use of array methods that don't mutate the array, e.g. `concat`, `filter`, `slice` instead of `push`, `pop`, `splice` this is because mutation may lead to unexpected and hard-to-debug issues, where data becomes incorrect somewhere, and you have no idea where it happens. [Read more](https://blog.sapegin.me/all/avoid-mutation/)


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

**Review by oxenprogrammer — APPROVED** [review]
Hi @nachosala89 and @Davidosky007 , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**